config/initializers/yaml_renderer.rb

```diff
@@ -1,4 +1,7 @@
 ActionController::Renderers.add :yaml do |obj, _|
-  data = JSON.load(obj.to_json).to_yaml
+  # Create a serializable resource instance
+  serializable_resource = ActiveModelSerializers::SerializableResource.new(obj)
+
+  data = serializable_resource.as_json.to_yaml
```

Comment:
you probably want `serializable_hash` here, but I think the better fix would just be

```diff
- ActionController::Renderers.add :yaml do |obj, _|
- data = JSON.load(obj.to_json).to_yaml
+ ActionController::Renderers.add :yaml do |obj, options|
+ data = data.serializable_hash(options) if data.respond_to?(:serializable_hash)
+ data = YAML.dump(data)
```